### PR TITLE
Retrieve historical data

### PR DESCRIPTION
**Pull request by numel007**
Routes = `/<ticker>/historical/<date_range>`

1. max
2. 1_year
3. 6_months
4. 1_month
5. 5_days

### CODE REVIEW

File: src/views.py [hunk 1]
```diff
@@ -73,79 +71,59 @@ def get_all(ticker):
         return new_data
 
 
-@app.route('/<ticker>/historical/max')
-def get_historical_all(ticker):
-    '''Retrieve all known historical data in 1 day increments for stock index'''
+@app.route('/<ticker>/historical/5_days')
```
Comment (numel007) [inline src/views.py]: Changes in `views.py` only involve rearranging the order of the routes in the code for easier readability, as well as removing code that was moved into `retrieve_historical()`

File: requirements.txt [hunk 2]
```diff
@@ -23,3 +23,5 @@ toml==0.10.2
 urllib3==1.26.3
 webencodings==0.5.1
 Werkzeug==1.0.1
+flask_api
```
Comment (pondorasti) [inline requirements.txt]: Did you use `pip freeze > requirements.txt`? The dependencies should also be locked to a version number.

File: src/scraper.py [hunk 3]
```diff
@@ -3,7 +3,44 @@
 from bs4 import BeautifulSoup as bs
 import html5lib
 from datetime import datetime, date
+from dateutil.relativedelta import relativedelta
 import csv
+import math
+
+
+def retrieve_historical(ticker, data_range):
```
Comment (pondorasti) [inline src/scraper.py]: If you are not using anything from the `Scraper` within this function you might be better off moving this helper function into a file of its own.

File: src/views.py [hunk 4]
```diff
@@ -67,4 +68,64 @@ def get_all(ticker):
...
+
+    data = retrieve_historical(ticker, '5_days')
+
+    if not data:
+        return f'{ticker} not found', status.HTTP_400_BAD_REQUEST
+
+    return jsonify(data)
+
+
+@app.route('/<ticker>/historical/1_month')
```
Comment (pondorasti) [inline src/views.py]: You could probably refactor those 5 routes into only `/<ticker>/historical/<duration>` and have a check that duration is valid before you pass it to `retrieve_historical()`

**Review by pondorasti — CHANGES_REQUESTED** [review]

File: src/scraper.py (hunk 3 above)
Comment (numel007) [reply src/scraper.py]: I was considering that but wasn't sure if it was necessary. I'll move it.

File: requirements.txt (hunk 2 above)
Comment (numel007) [reply requirements.txt]: [Resolved](https://github.com/BoringStocks/StonkScraper/pull/12/commits/1b30786f950f3bbe0587f87095855f260749dacc)

File: src/scraper.py (hunk 3 above)
Comment (numel007) [reply src/scraper.py]: [Resolved](https://github.com/BoringStocks/StonkScraper/pull/12/commits/f71bf7579ac443ac7771bfa9f626cba4a4d0a28e)

File: requirements.txt (hunk 2 above)
Comment (pondorasti) [reply requirements.txt]: Why doesn't `flask_api` have a version number?

Comment (numel007) [reply requirements.txt]: 😒 Don't ask me how I missed that, fixing now

File: src/views.py (hunk 4 above)
Comment (numel007) [reply src/views.py]: [Resolved](https://github.com/BoringStocks/StonkScraper/pull/12/commits/5a50a171fff66bf2c7fe15e9a579086e56dd892e)

File: requirements.txt (hunk 2 above)
Comment (numel007) [reply requirements.txt]: [Resolved for the final time](https://github.com/BoringStocks/StonkScraper/pull/12/commits/1733617c5629a5fe89cc7b29245ac3c157295121)

**Review by pondorasti — APPROVED** [review]
Looks great! Good job!

**Pull request merged**